src/app/trip-page/trip-overview/trip-resolver.component.ts

```diff
@@ -2,16 +2,34 @@ import { Injectable } from '@angular/core';
 import { ActivatedRouteSnapshot, Resolve, Router } from '@angular/router';
 import { TripService } from '../../trips/.';
 import { take } from 'rxjs';
+import { ITrip } from 'src/app/shared';
+import { AuthService } from 'src/app/user';
 
 @Injectable()
 export class TripResolver implements Resolve<any> {
-  constructor(private tripService: TripService, private router: Router) {}
+  constructor(
+    private tripService: TripService,
+    private router: Router,
+    private auth: AuthService
+  ) {}
+
+  // verify if the user is a member of the trip
+  checkUserIsMemberOfTrip(trip: ITrip): boolean {
+    for (let member of trip.members) {
+      if (member.username === this.auth.currentUser.username) return true;
```

Comment:
You should get in the habit of always using blocks even for 1 line if conditions. It might be more code but it's more readable and one day it'll save you a bug :)